File: Greedy/큰 수의 법칙/큰 수의 법칙.swift
import Foundation

// 그리디 적용 전
let input = readLine()!.components(separatedBy: " ").map { Int(String($0)) ?? 0}
let (n, m, k) = (input[0], input[1], input[2])
var number = readLine()!.components(separatedBy: " ").map { Int(String($0)) ?? 0}

number.sort(by: >)
var result = 0

for i in 1..<m+1 {
    if k % i == 0 {
        result += number[1]
    } else {
        result += number[0]
    }
}

print(result)

// 그리디 적용 후
let input = readLine()!.components(separatedBy: " ").map { Int(String($0)) ?? 0}
let (n, m, k) = (input[0], input[1], input[2])
var number = readLine()!.components(separatedBy: " ").map { Int(String($0)) ?? 0}

number.sort(by: >)
let first = number[0] // 가장 큰 수
let second = number[1] // 두 번째로 큰 수

// 가장 큰 수가 더해지는 횟수 계산
var count = Int(m / (k+1)) * k
count += m % (k+1)

var result = 0
result += count * first
result += (m-count) * second

print(result)

File: Greedy/큰 수의 법칙/큰 수의 법칙.py
## 그리디 적용 전
n, m, k = map(int, input().split())
number = list(map(int, input().split()))
    
answer = 0
number.sort()
    
for i in range(1, m+1):
    if k % i == 0:
        answer += number[-2]
    else:
        answer += number[-1]
    
print(answer)

## 그리디 적용 후
n, m, k = map(int, input().split())
data = list(map(int, input().split()))

data.sort()
first = data[- 1] # 가장 큰 수
second = data[- 2] # 두 번째로 큰 수

count = int(m / (k+1)) * k
count += m % (k+1)

result = 0
result += (count) * first
result += (m - count) * second

print(result)
